R/shift.R

```diff
@@ -9,7 +9,7 @@ shift = function(x, n=1L, fill=NA, type=c("lag", "lead", "shift"), give.names=FA
       else nx = paste0("V", if (is.atomic(x)) 1L else seq_along(x))
     }
     else nx = names(x)
-    if (type!="shift") {
+    if (!(type %chin% c("shift", "cyclic"))) {
```

Comment:
should we error (warn?) if `fill` is no missing for cyclic?

Reply:
I also thought about a warning (thinking again about it, error seems a better choice) if both `fill` with an argument and `type="cyclic"` are used, but then I gave up the thought, because for me there is no ambiguity of how cyclic should work other than wrap around like Python's `numpy.roll`.

Reply:
agree it's pretty unambiguous, I am thinking of a programmatic edge case where the user inadvertently supplies both in a loop, etc. a warning can alert them to possibly unintended behavior from their code. (thinking again a warning is definitely better than an error)